tensorflow/lite/micro/kernels/unidirectional_sequence_lstm.cc

```diff
@@ -1266,6 +1266,14 @@ TfLiteStatus UnidirectionalSequenceLstmPrepare(TfLiteContext* context,
   TF_LITE_ENSURE_EQ(context, NumElements(output_state), n_batch * n_output);
   TF_LITE_ENSURE_EQ(context, NumElements(cell_state), n_batch * n_cell);
 
+  // Check the shape of output tensor against that of input tensor
+  TF_LITE_ENSURE_EQ(context, input->dims->size, output->dims->size);
```

Comment:
Could you please elaborate this a bit? I am not entirely following the reason behind this. 

input->dims->size == output-dims->size
and the first N element of input->dims->data and output->dims->data are the same.
Then the last element of output->dims->data is the same as n_output.

However, it seems n_output = input->dims->data[2].
So, is input->dims->size guaranteed to be 3 or something? 
If yes, check do not need loop.
If no,  this guaratee should be called out.

WDYT?

Reply:
In this implementation it does fix the dims->size to be 3 and directly assign input->dims->data[2] to n_input.

I was trying to follow the logic in the TFLite implementation as close as possible:

  // Resize the output tensors.
  TfLiteIntArray* output_size = TfLiteIntArrayCopy(input->dims);
  output_size->data[input->dims->size - 1] = n_output;
  TF_LITE_ENSURE_OK(context,
                    context->ResizeTensor(context, output, output_size));

Of course we do not do resizing and just check the dimensions. 

I kind of think the loop makes it more generic and extensible in case a higher dimension is added in the future, in comparison with the following:

TF_LITE_ENSURE_EQ(context, input->dims->data[0], output->dims->data[0]);
TF_LITE_ENSURE_EQ(context, input->dims->data[1], output->dims->data[1]);

Please let me know your thoughts on this. Thanks for reviewing it promptly.

Reply:
I might prefer the simpler approach here just to avoid the confusion for future code readers. But if you really like the more general way, that is fine too (but please add the explanation in the code to address potential confusion like mine).

Reply:
Changed to the simpler approach and added a check to make sure the dimension size is 3.